### PR TITLE
Consider possible schema for countdown object

### PR DESCRIPTION
**Pull request by ConcurrencyPractitioner**
Lol why not

### CODE REVIEW

**Comment by wilsonh1** [comment]
Should probably use [primitive arrays](https://mongoosejs.com/docs/schematypes.html#arrays) instead of embedded documents.

**Comment by ConcurrencyPractitioner** [comment]
Alright, I changed it to map instead. Seems better after some thinking.

**Comment by ConcurrencyPractitioner** [comment]
@wilsonh1 Got the process together:

MongooseMap has two primary methods: get and set.
When the game begins, each time a person joins, we put them in the "people" map and set their score to zero. When that map size reaches the countdown group size, we begin the game.

We should consider the deployment of daemon threads (admittedly a ton more complex but makes available to us a lot of more complex features). But we can do some simpler stuff too. When the first person answers, we block all other people from answering, and announce that this person won.

The second "problems" map will be used to check if a certain problem has been answered correctly. Idk, just some guess work here.

**Comment by wilsonh1** [comment]
Note that mongoose queries/updates are async. `joinCountdown(senderId, gameId)` in line 45 will probably execute before `Countdown.create()` is finished.

Also, would prefer if the schema is kept in `models`.

File: countdown.js [hunk 1]
```diff
@@ -21,67 +19,83 @@ var CountdownSchema = new Schema ({
...
-    ftw.sendMessage(senderId, {test: "Joined the game successfully"}, false);
-    if (currentCountdown.currentCount == currentCountdown.count) {
-	launchGame();
+// TODO: join if this thing has not launched
+function joinIfNotLaunched(gameId) {
+    
+}
+
+function joinCountdown(senderId, gameId) {
+    User.updateOne({user_id: senderId}, {game_id: id}, function (err, res) {
```
Comment (wilsonh1) [inline countdown.js]: Use `upsert` to create a document if there isn't one already.

File: ftw.js [hunk 2]
```diff
@@ -7,6 +7,31 @@ var db = mongoose.connect(process.env.MONGODB_URI, {useNewUrlParser: true, useUn
...
+                return undefined;
+            }
+            var rand = Math.floor(Math.random() * res);
+
+            var pQ = Problem.findOne({p_id: rand}).select({statement: 1, image: 1, _id: 0}).lean();
+            pQ.exec(function(errP, pObj) {
+                if (errP)
+                    console.log(errP);
+                else {
+		    return pObj;
```
Comment (wilsonh1) [inline ftw.js]: This won't actually return an object; it'll only exit the callback function.

Comment (ConcurrencyPractitioner) [reply ftw.js]: Yea, I realized. I was being dumb.

File: countdown.js (hunk 1 above)
Comment (ConcurrencyPractitioner) [reply countdown.js]: Cool, yeah, I saw it. {upsert:true} right?

**Comment by ConcurrencyPractitioner** [comment]
@wilsonh1  Rough draft is done. Will furnish it and you can probably merge soon.

**Comment by ConcurrencyPractitioner** [comment]
@wilsonh1 Alright, done.

**Comment by wilsonh1** [comment]
Have you tested this yet? [Here's](https://www.sitepoint.com/building-facebook-chat-bot-node-heroku/) a guide for setting up a messenger bot with heroku.

**Comment by ConcurrencyPractitioner** [comment]
Nope, haven't yet.

**Comment by ConcurrencyPractitioner** [comment]
Alright, guarantee compile. Beyond that, who knows if it actually works.

**Comment by ConcurrencyPractitioner** [comment]
Note: This PR will be not be merged to this directory. Will be added to https://github.com/wilsonh1/ClubSignIn

**Pull request closed without merging**